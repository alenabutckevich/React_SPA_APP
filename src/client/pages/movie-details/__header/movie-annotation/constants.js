export const BASE_CLASS = 'movie_annotation';