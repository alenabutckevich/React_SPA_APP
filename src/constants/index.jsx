export const START_FETCHING_MOVIES_BY_TITLE = 'START_FETCHING_MOVIES';
export const FETCH_MOVIES_BY_TITLE_SUCCESS = 'FETCH_MOVIES_SUCCESS';
export const FETCH_MOVIES_BY_TITLE_FAILURE = 'FETCH_MOVIES_FAILURE';
export const START_FETCHING_MOVIES_BY_DIRECTOR = 'START_FETCHING_MOVIES';
export const FETCH_MOVIES_BY_DIRECTOR_SUCCESS = 'FETCH_MOVIES_SUCCESS';
export const FETCH_MOVIES_BY_DIRECTOR_FAILURE = 'FETCH_MOVIES_FAILURE';
export const START_FETCHING_MOVIE_BY_ID = 'START_FETCHING_MOVIE_BY_ID';
export const FETCH_MOVIE_BY_ID_SUCCESS = 'FETCH_MOVIE_BY_ID_SUCCESS';
export const FETCH_MOVIE_BY_ID_FAILURE = 'FETCH_MOVIE_BY_ID_FAILURE';
export const START_FETCHING_MOVIE_CAST_AND_CREW = 'START_FETCHING_MOVIE_CAST_AND_CREW';
export const FETCH_MOVIE_CAST_AND_CREW_SUCCESS = 'FETCH_MOVIE_CAST_AND_CREW_SUCCESS';
export const FETCH_MOVIE_CAST_AND_CREW_FAILURE = 'FETCH_MOVIE_CAST_AND_CREW_FAILURE';
export const SET_SORT_FILTER = 'SET_SORT_FILTER';
export const SET_SEARCH_FILTER = 'SET_SEARCH_FILTER';
export const SET_SEARCH_QUERY = 'SET_SEARCH_QUERY';
export const RESET_CURRENT_MOVIE = 'RESET_CURRENT_MOVIE';
export const START_FETCHING_GENRES = 'START_FETCHING_GENRES';
export const FETCH_GENRES_SUCCESS = 'FETCH_GENRES_SUCCESS';
export const FETCH_GENRES_FAILURE = 'FETCH_GENRES_FAILURE';
