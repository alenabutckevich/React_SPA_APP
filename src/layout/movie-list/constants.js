export const RELEASE_DATE_FILTER = {
    name: 'release date',
    value: 'release_date'
}

export const RATING_FILTER = {
    name: 'rating',
    value: 'vote_average'
}

export const TITLE_FILTER = {
    name: 'title',
    value: 'title'
}