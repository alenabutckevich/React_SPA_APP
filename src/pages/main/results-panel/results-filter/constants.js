export const RELEASE_DATE_FILTER = 'release date';
export const RATING_FILTER = 'rating';