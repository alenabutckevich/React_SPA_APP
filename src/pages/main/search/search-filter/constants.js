export const TITLE_FILTER = 'title';
export const DIRECTOR_FILTER = 'director';