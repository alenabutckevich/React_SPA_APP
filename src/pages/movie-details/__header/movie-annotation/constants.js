export const BASE_CLASS = 'movie-annotation';