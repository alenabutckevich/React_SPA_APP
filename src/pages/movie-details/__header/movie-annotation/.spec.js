import React from 'react';
import toJson from 'enzyme-to-json';
import MovieAnnotation from './';
import { BASE_CLASS } from './constants';

const setup = propOverrides => {
    const { runtime, genres } = propOverrides;
    const props = Object.assign({
        movie: {
            poster_path: 'path',
            title: 'title',
            vote_average: '9.9',
            genres: genres,
            release_date: '12-12-2012',
            runtime: runtime,
            overview: 'overview'
        },
        cast: 'cast',
        director: 'director',
    }, propOverrides)

    const wrapper = shallow(<MovieAnnotation {...props} />)

    return {
        props,
        wrapper
    }
}

describe('<MovieAnnotation/>', () => {
    const testMovieAnnotationRender = (runtime, genres) => {
        it('render movie annotation', () => {
            const { wrapper } = setup({ runtime: runtime, genres: genres });

            expect(wrapper.find(`.${BASE_CLASS}__image`).prop('src')).toEqual('https://image.tmdb.org/t/p/w500/path');
            expect(wrapper.find(`.${BASE_CLASS}__title`).text()).toEqual('title');
            expect(wrapper.find(`.${BASE_CLASS}__rating`).text()).toEqual('9.9');
            expect(wrapper.find(`.${BASE_CLASS}__genre`).text()).toEqual(genres.length ? 'genre1' : '');
            expect(wrapper.find(`.${BASE_CLASS}__release-date`).text()).toEqual('2012');
            expect(wrapper.find(`.${BASE_CLASS}__duration`).text()).toEqual(`${runtime || 0} min`);
            expect(wrapper.find(`.${BASE_CLASS}__description`).text()).toEqual('overview');
            expect(wrapper.find(`.${BASE_CLASS}__director`).text()).toEqual('Director: director');
            expect(wrapper.find(`.${BASE_CLASS}__cast`).text()).toEqual('Cast: cast');

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    }

    testMovieAnnotationRender('100', [{id: 1, name: 'genre1'}]);
    testMovieAnnotationRender('', []);
})
