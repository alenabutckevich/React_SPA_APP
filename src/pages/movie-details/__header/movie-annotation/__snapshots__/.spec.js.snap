// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieAnnotation/> render movie annotation 1`] = `
<div
  className="movie-annotation"
>
  <img
    className="movie-annotation__image"
    src="https://image.tmdb.org/t/p/w500/path"
  />
  <h2
    className="movie-annotation__title"
  >
    title
  </h2>
  <button
    className="movie-annotation__rating"
    disabled={true}
  >
    9.9
  </button>
  <div
    className="movie-annotation__genre"
  >
    genre1
  </div>
  <div
    className="movie-annotation__release-date"
  >
    2012
  </div>
  <div
    className="movie-annotation__duration"
  >
    100 min
  </div>
  <h2
    className="movie-annotation__description"
  >
    overview
  </h2>
  <div
    className="movie-annotation__director"
  >
    Director: 
    director
  </div>
  <div
    className="movie-annotation__cast"
  >
    Cast: 
    cast
  </div>
</div>
`;

exports[`<MovieAnnotation/> render movie annotation 2`] = `
<div
  className="movie-annotation"
>
  <img
    className="movie-annotation__image"
    src="https://image.tmdb.org/t/p/w500/path"
  />
  <h2
    className="movie-annotation__title"
  >
    title
  </h2>
  <button
    className="movie-annotation__rating"
    disabled={true}
  >
    9.9
  </button>
  <div
    className="movie-annotation__genre"
  />
  <div
    className="movie-annotation__release-date"
  >
    2012
  </div>
  <div
    className="movie-annotation__duration"
  >
    0 min
  </div>
  <h2
    className="movie-annotation__description"
  >
    overview
  </h2>
  <div
    className="movie-annotation__director"
  >
    Director: 
    director
  </div>
  <div
    className="movie-annotation__cast"
  >
    Cast: 
    cast
  </div>
</div>
`;
